src/workerd/api/basics.c++

```diff
@@ -669,6 +712,97 @@ void AbortSignal::triggerAbort(
   dispatchEventImpl(js, js.alloc<Event>(kj::str("abort")));
 }
 
+void AbortSignal::serialize(jsg::Lock& js, jsg::Serializer& serializer) {
+  auto& handler = JSG_REQUIRE_NONNULL(serializer.getExternalHandler(), DOMDataCloneError,
+      "AbortSignal can only be serialized for RPC.");
+  auto externalHandler = dynamic_cast<RpcSerializerExternalHander*>(&handler);
+  JSG_REQUIRE(
+      externalHandler != nullptr, DOMDataCloneError, "AbortSignal can only be serialized for RPC.");
+
+  serializer.writeRawUint32(static_cast<uint>(canceler->isCanceled()));
+  serializer.writeRawUint32(static_cast<uint>(flag));
+  KJ_IF_SOME(r, reason) {
+    serializer.write(js, r.getHandle(js));
+  } else {
+    serializer.write(js, js.undefined());
+  }
+
+  auto& ioContext = IoContext::current();
+
+  auto streamCap =
+      externalHandler->writeStream([&](rpc::JsValue::External::Builder builder) mutable {
+    auto field = builder.initAbortSignal();
+    field.setSignal(kj::heap<AbortSignalImpl>(ioContext, JSG_THIS));
```

Comment:
There's one part I still have a mental block on - how do I change the code to avoid passing the capability the wrong way around?

Reply:
@kentonv Do I just get rid of these two lines?

Reply:
Change `JsValue.External.abortSignal` to have type `Void`. Then you do `builder.setAbortSignal()` here to set it, but it's just empty, because there's nothing you need to send here.